src/app/clusters/icd-management-server/icd-management-server.cpp

```diff
@@ -137,76 +149,211 @@ CHIP_ERROR IcdManagementAttributeAccess::ReadRegisteredClients(EndpointId endpoi
 
 CHIP_ERROR IcdManagementAttributeAccess::ReadICDCounter(EndpointId endpoint, AttributeValueEncoder & encoder)
 {
-    return encoder.Encode(ICDManagementServer::GetInstance().GetICDCounter());
+    return encoder.Encode(mICDData->GetICDCounter());
 }
 
 CHIP_ERROR IcdManagementAttributeAccess::ReadClientsSupportedPerFabric(EndpointId endpoint, AttributeValueEncoder & encoder)
 {
-    return encoder.Encode(ICDManagementServer::GetInstance().GetClientsSupportedPerFabric());
+    return encoder.Encode(mICDData->GetClientsSupportedPerFabric());
 }
 
-CHIP_ERROR CheckAdmin(chip::app::CommandHandler * commandObj, const chip::app::ConcreteCommandPath & commandPath, bool & is_admin)
+/**
+ * @brief Implementation of Fabric Delegate for ICD Management cluster
+ */
+class IcdManagementFabricDelegate : public FabricTable::Delegate
 {
-    RequestPath requestPath{ .cluster = commandPath.mClusterId, .endpoint = commandPath.mEndpointId };
-    CHIP_ERROR err = GetAccessControl().Check(commandObj->GetSubjectDescriptor(), requestPath, Privilege::kAdminister);
-    if (CHIP_NO_ERROR == err)
-    {
-        is_admin = true;
-    }
-    else if (CHIP_ERROR_ACCESS_DENIED == err)
+public:
+    void Init(PersistentStorageDelegate & storage, Crypto::SessionKeystore * sessionKeystore, ICDData & icdData)
```

Comment:
Again, is this a SessionKeystore or SymmetricKeystore?

Reply:
renamed all the sessionKeystore to symmetricKeystore.